src/troute_model.py

```diff
@@ -231,8 +264,27 @@ def run(self, values: dict, until=300):
         
         # update model time
         self._time += self._time_step * nts
+        
+        if self.showtiming:
+            output_end_time = time.time()                           # end of recording time for creating output #
+            self.task_times['output_time'] += output_end_time - run_end_time
 
-
+    def finalize(self,):
+            if self.showtiming:
+                print('***************** TIMING SUMMARY *****************')
+                print('----------------------------------------')
+                total_time = (round(self.task_times['network_time'], 2) + 
+                              round(self.task_times['data_assimilation_time'], 2) + 
+                              round(self.task_times['forcing_time'], 2) + 
+                              round(self.task_times['run_time'], 2) + 
+                              round(self.task_times['output_time'], 2)
+                              )
+                for key in self.task_times:
+                    time_value = self.task_times[key]
+                    percentage = (time_value / total_time) * 100
+                    print(
+                    f'{key} construction: {time_value:.2f} secs, {percentage:.2f} %'
+                )
 # Utility functions -------
 def _read_config_file(custom_input_file): #TODO: Update this function, I dont' think
```

Comment:
@AminTorabi-NOAA  Need to return _log_parameters from this function.

Reply:
Sure.

Reply:
Sorry, maybe I selected the wrong line. We don't need log_parameters returned from the finalize() function, we need it returned from _read_config_file()

Reply:
Oh thats True. Thanks